src/main/java/uk/gov/companieshouse/insolvency/delta/processor/InsolvencyDeltaProcessor.java

```diff
@@ -73,6 +77,13 @@ public void processDelta(Message<ChsDelta> chsDelta,
 
         Insolvency insolvency = insolvencyDelta.getInsolvency().get(0);
 
+        try {
+            validator.validateCaseDates(insolvency);
+        } catch (Exception error) {
+            throw new NonRetryableErrorException("Error when validating case dates: "
```

Comment:
The validator already throws a non retryable error do we need to catch and rethrow?

Reply:
Yes, CaseNumber.class.getMethod and getter.invoke(caseNumber) throw IOExceptions, this top level catch is for any exceptions that occur accessing these methods.